lib/config.js

```diff
@@ -163,13 +163,12 @@ internals.routeBase = Joi.object({
         modify: Joi.boolean(),
         options: Joi.object().default(),
         ranges: Joi.boolean().default(true),
-        sample: Joi.number().min(0).max(100),
+        sample: Joi.number().min(0).max(100).when('modify', { is: true, then: Joi.forbidden() }),
         schema: Joi.alternatives(Joi.object(), Joi.array(), Joi.func()).allow(true, false),
         status: Joi.object().pattern(/\d\d\d/, Joi.alternatives(Joi.object(), Joi.array(), Joi.func()).allow(true, false))
     })
         .default()
-        .without('modify', 'sample')
-        .assert('options.stripUnknown', Joi.when('modify', { is: true, otherwise: Joi.forbidden() }), 'meet requirement of having peer modify set to true'),
+        .assert('options.stripUnknown', Joi.when('modify', { is: true, otherwise: [Joi.boolean().valid(false), Joi.forbidden()] }), 'meet requirement of having peer modify set to true'),
```

Comment:
Do you need to keep the forbidden as alternative?

Reply:
I just keep the existing one while allowing "false" value.

Do you want me to remove the forbidden as an alternative here?

Reply:
Actually, this is correct.

Reply:
It's not needed because the default is optional.